mne/preprocessing/nirs/_scalp_coupling_index.py

```diff
@@ -59,7 +59,10 @@ def scalp_coupling_index(raw, l_freq=0.7, h_freq=1.5,
 
     sci = np.zeros(picks.shape)
     for ii in picks[::2]:
-        c = np.corrcoef(filtered_data[ii], filtered_data[ii + 1])[0][1]
+        with np.errstate(invalid='ignore'):
+            c = np.corrcoef(filtered_data[ii], filtered_data[ii + 1])[0][1]
+        if not np.isfinite(c):  # someone had std=0
```

Comment:
Does this mean that the channel is flat? I.e. it's all the same value? If this is the case would it not be better to throw the user a warning and tell them to set it to bad?

Reply:
Warning could be useful, yeah. This was mostly just laziness since I think/hope the outcome of the algorithm is still to clearly say the channel is bad anyway (e.g., with a SCI of 0). Maybe I should assert this in a test, then a warning isn't necessary? Or maybe we need both warning and sci=0?

Reply:
You're right, as long as it comes out with a 0 value the result will be the same to the user.

We should probably have a flat signal check as a seperate function check. I think I have seen that around somewhere in MNE already, Ill look.

Reply:
Yes we have a flat marking / annotating function

https://mne.tools/stable/generated/mne.preprocessing.annotate_flat.html